### PR TITLE
Popup window to display new matches

### PR DESCRIPTION
**Pull request by hadleyshapland**
image

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/MatchServlet.java [hunk 1]
```diff
@@ -39,11 +39,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public synchronized void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
 
-    String[] matchesToSave = request.getParameterValues("new-matches");
+    if(request.getParameterValues("new-matches") != null) {
```
Comment (ghandre) [inline src/main/java/com/google/sps/servlets/MatchServlet.java:42]: Could you log or throw an error if "new-matches" is not present?

File: src/main/webapp/logged_in_homepage.html [hunk 2]
```diff
@@ -1,109 +1,103 @@
...
-
-  <!-- Bootstrap core JavaScript -->
-  <script src="vendor/jquery/jquery.min.js"></script>
-  <script src="vendor/bootstrap/js/bootstrap.bundle.min.js"></script>
-
-
+      <!-- /.container -->
+    </footer>
+    <!-- Bootstrap core JavaScript -->
+    <script src="vendor/jquery/jquery.min.js"></script>
```
Comment (ghandre) [inline src/main/webapp/logged_in_homepage.html:99]: Could you move these script includes to the top with the others?

File: src/main/java/com/google/sps/servlets/MatchServlet.java (hunk 1 above)
Comment (jerrychangusa) [inline src/main/java/com/google/sps/servlets/MatchServlet.java:42]: maybe include a test that would fail without this null check?

File: src/main/webapp/logged_in_homepage.html [hunk 3]
```diff
@@ -1,109 +1,103 @@
...
+    <meta name="author" content="Samuel Sandoval and Hadley Shapland">
+    <title>Homepage</title>
+    <!-- Bootstrap core CSS -->
+    <link href="vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">
+    <!-- Custom fonts for this template -->
+    <link href="https://fonts.googleapis.com/css?family=Catamaran:100,200,300,400,500,600,700,800,900" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Lato:100,100i,300,300i,400,400i,700,700i,900,900i" rel="stylesheet">
+    <!-- Custom styles for this template -->
+    <link href="css/one-page-wonder.min.css" rel="stylesheet">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.10.0/css/bootstrap-select.min.css" rel="stylesheet" />
```
Comment (jerrychangusa) [inline src/main/webapp/logged_in_homepage.html:16]: nit: remove closing / here.

File: src/main/webapp/logged_in_homepage.html [hunk 4]
```diff
@@ -1,109 +1,103 @@
...
+            <h3>Get New Matches</h3>
+            <label for="product_area"><b>Product Area</b></label>
+            <input
+              type="text"
+              placeholder="Enter Product Area"
+              id="product_area"
+              required
+              />
+            <input type="button" class="btn btn-primary" value="Submit" onclick="sendMatchRequest()" data-toggle="modal" data-target="#exampleModal"/>
+          </form>
```
Comment (samuelsandoval1) [inline src/main/webapp/logged_in_homepage.html:61]: nit: No closing brace for input needed. L60

File: src/main/java/com/google/sps/servlets/MatchServlet.java (hunk 1 above)
Comment (hadleyshapland) [reply src/main/java/com/google/sps/servlets/MatchServlet.java:42]: I added an error message

File: src/main/webapp/logged_in_homepage.html (hunk 2 above)
Comment (hadleyshapland) [reply src/main/webapp/logged_in_homepage.html:99]: I might be wrong about this so let me know, but I thought that the JavaScript includes were supposed to go at the bottom of the .html?

File: src/main/java/com/google/sps/servlets/MatchServlet.java (hunk 1 above)
Comment (hadleyshapland) [reply src/main/java/com/google/sps/servlets/MatchServlet.java:42]: Good idea, I added one!

File: src/main/webapp/logged_in_homepage.html (hunk 3 above)
Comment (hadleyshapland) [reply src/main/webapp/logged_in_homepage.html:16]: Done

File: src/main/webapp/logged_in_homepage.html (hunk 4 above)
Comment (hadleyshapland) [reply src/main/webapp/logged_in_homepage.html:56]: Done

File: src/main/webapp/logged_in_homepage.html (hunk 2 above)
Comment (jerrychangusa) [reply src/main/webapp/logged_in_homepage.html:94]: From what I understand, script tags can be included in either the <head> element or at the bottom of the <body> element. The advantage of placing it at the bottom is it improves the display speed (script interpretation slows down the display).
